twitter-clone/app/src/main/java/io/github/vnicius/twitterclone/ui/main/MainActivity.kt

```diff
@@ -42,54 +44,66 @@ class MainActivity : AppCompatActivity(), MainContract.View, View.OnClickListene
 
                 startActivity(intent)
             }
+
+            btn_connection_error_action.id -> {
+                presenter.getTrends()
+            }
         }
     }
 
     override fun showLoader() {
-        changeFragment(LoaderFragment.newInstance())
+        hideContent()
+        inc_main_spinner.visibility = View.VISIBLE
     }
 
     override fun showTrends(trends: Array<Trend>) {
-        val fragment = TrendsFragment.newInstance()
-        val args = Bundle()
-
-        // pass the list of trends to the Trend Fragment by argument
-        args.putSerializable(TrendsFragment.ARG_CODE, trends as Serializable)
-        fragment.arguments = args
-
-        changeFragment(fragment)
+        hideContent()
+        ll_main_trends.visibility = View.VISIBLE
+
+        rv_main_trends_list.apply {
```

Comment:
Maybe you could move the entire code inside this `apply` to a method like `setupRecyclerView()` and call it just once in the `Activity.onCreate()`. From what I see this `showTrends()` method just need to call `hideContent()` and change the visibility of the `ll_main_trends`, am I right?

Reply:
I think in this case you will also need to update the data in the adapter and call the `notifyDataSetChanged()`